lib/resources/user-collection.js

```diff
@@ -166,6 +138,78 @@ UserCollection.prototype.handle = function (ctx) {
   }
 };
 
+function handleLogin(uc, ctx) {
+  var path = uc.path
+    , credentials = ctx.req.body || {};
+  
+  debug('trying to login as %s', credentials.username);
+  
+  uc.store.first({ username: credentials.username }, function (err, user) {
+    if (err) return ctx.done(err);
+    // keep a clone of the user so we can compare it later to see if any changes were made in the login event
+    var userClone = user ? _.clone(user) : null
+      , domain = { 'me': userClone, 'data': userClone, 'success': false };
+    
+    // checks if the user was changed in the login event and saves it if it was
+    function saveUser(fn) {
```

Comment:
I'm not sure moving loginDone, loginFail and saveUser is right for this purpose. 

If I just put uc.store.first({ username: credentials.username }) in a prototype, then you can override that prototype and chance it for your needs in one place, and it's simpler.

Do you have other requirements this wouldn't satisfy?